lib/faraday_middleware/gzip.rb

```diff
@@ -48,7 +48,17 @@ def uncompress_gzip(body)
     end
 
     def inflate(body)
+      # Inflate as a DEFLATE (RFC 1950+RFC 1951) stream
       Zlib::Inflate.inflate(body)
+    rescue Zlib::DataError
```

Comment:
Unfortunately the Zlib module does not expose `inflateGetHeader()`, but even if it did we'd end up having to rescue Zlib::StreamError because the raw stream does not have a header.
You could of course check for the first few bytes for magic numbers (in the case of zlib/inflate it's variable bit patterns) but I'd rather leave it to Zlib.